pkg/build/oci/oci.go

```diff
@@ -68,6 +68,26 @@ var keychain = authn.NewMultiKeychain(
 	github.Keychain,
 )
 
+var remoteOpts []remote.Option
+
+func init() {
+	remoteOpts = []remote.Option{remote.WithAuthFromKeychain(keychain)}
+
+	pusher, err := remote.NewPusher(remoteOpts...)
+	if err == nil {
+		remoteOpts = append(remoteOpts, remote.Reuse(pusher))
+	} else {
+		log.DefaultLogger().Infof("NewPusher(): %v", err)
+	}
+
+	puller, err := remote.NewPuller(remoteOpts...)
```

Comment:
I'm probably dumb, but why isn't a `pusher` also a `puller`, so we can create just one and `Reuse` it for both pushes and pulls?

Reply:
Because of how many (including ours) registries implemented auth incorrectly, but also because they're structs.